src/ResponseTagger.php

```diff
@@ -36,10 +50,31 @@ class ResponseTagger
      * Constructor
      *
      * @param TagsInterface $client
+     * @param array         $options supported options:
+     *                               - strict (bool) Default: false. If set to true, throws exception when adding empty tags.
      */
-    public function __construct(TagsInterface $client)
+    public function __construct(TagsInterface $client, array $options = array())
     {
         $this->client = $client;
+
+        if (!isset(self::$resolver)) {
+            self::$resolver = new OptionsResolver();
+            $this->configureOptions(self::$resolver);
+        }
+
+        $this->options = self::$resolver->resolve($options);
+    }
+
+    /**
+     * Configure options
+     *
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
```

Comment:
> so someone can override the method in an extend?

then protected would be enough. i'd move that to the constructor for
now. if somebody comes up with a good reason why they want to extend it,
we can still refactor to a protected method without causing any BC break
so they can extend it.